src/common/uilocale.cpp

```diff
@@ -275,6 +275,329 @@ wxLocaleIdent wxLocaleIdent::FromTag(const wxString& tag)
+    wxString supportedScriptIn;
+    wxString supportedRegionIn;
+    wxString supportedLanguage;
+    wxString supportedScript;
+    wxString supportedRegion;
+    int supportedMinDistance;
+    bool directMatch = false;
+
+    // Loop over desired languages
+    for (size_t j = 0; !directMatch && j < desired.size(); ++j)
+    {
+        // Determine maximized desired locale identifiers
+        desiredLocaleIn = wxLocaleIdent::FromTag(desired[j]);
+        desiredLocaleMax = AddLikelySubtags(desiredLocaleIn);
+        desiredLanguage = desiredLocaleMax.GetLanguage();
+        desiredScript = desiredLocaleMax.GetScript();
+        desiredRegion = desiredLocaleMax.GetRegion();
+
+        // Loop over supported languages
+        for (size_t k = 0; !directMatch && k < supported.size(); ++k)
+        {
+            // Languages closer to the beginning of the list of desired languages are preferred
+            int distance = j * defaultDemotion;
+
+            // Check direct match
+            if (desired[j].IsSameAs(supported[k], false))
+            {
+                // This is the best match we can find, so leave the loop
+                if (distance < bestDistance)
+                {
+                    bestDistance = distance;
+                    bestDesired = j;
+                    bestSupported = k;
+                }
+                directMatch = true;
+                break;
+            }
+
+            // No direct match: Slightly increase distance
+            ++distance;
+
+            // Determine maximized supported locale identifiers
+            if (j == 0)
+            {
+                supportedLocaleIn = wxLocaleIdent::FromTag(supported[k]);
+                supportedLocalesIn.push_back(supportedLocaleIn);
+                supportedLocalesMax.push_back(AddLikelySubtags(supportedLocaleIn));
+                wxLocaleIdent supportedLocaleMin = RemoveLikelySubtags(supportedLocaleIn);
+                int minDistance = 0;
+                if (!supportedLocaleMin.GetScript().empty()) minDistance += 1;
+                if (!supportedLocaleMin.GetRegion().empty()) minDistance += 1;
+                supportedLocalesMinDistance.push_back(minDistance);
+            }
+            supportedLanguageIn = supportedLocalesIn[k].GetLanguage();
+            supportedScriptIn = supportedLocalesIn[k].GetScript();
+            supportedRegionIn = supportedLocalesIn[k].GetRegion();
+            supportedLanguage = supportedLocalesMax[k].GetLanguage();
+            supportedScript = supportedLocalesMax[k].GetScript();
+            supportedRegion = supportedLocalesMax[k].GetRegion();
+            supportedMinDistance = supportedLocalesMinDistance[k];
+
+            // Check for matching language, script and region
+            if (!directMatch && !desiredLanguage.empty() && !supportedLanguage.empty())
+            {
+                // Check language
+                if (supportedLanguage.IsSameAs(desiredLanguage, false))
```

Comment:
Minor, but I find

```suggestion
                if (supportedLanguage.CmpNoCase(desiredLanguage) == 0)
```

much better (or, as already stated above, boolean parameters are really unreadable).

Reply:
Ok, I will change the code accordingly.